UnityProject/Assets/Scripts/Variable Viewer/BookViewer/ElementDisplay/ElementTypes/Vectors/GUI_P_Vectors.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+using UnityEngine.UI;
+using DatabaseAPI;
+
+/// <summary>
+/// Used to display vectors in the Variable viewer and to do all the Serialisation and deserialisation of vectors
+/// </summary>
+public class GUI_P_Vectors : PageElement
+{
+	public InputField INX;
+	public InputField INY;
+	public InputField INZ;
+
+	public bool IsSentence;
+	public bool iskey;
+
+	public Vector IsThisVector;
```

Comment:
If these are only used in this class you should switch them to private and use [SerializeField].
```suggestion
	[SerializeField]
	private InputField INX;
	[SerializeField]
	private InputField INY;
	[SerializeField]
	private InputField INZ;

	[SerializeField]
	private bool isSentence;
	[SerializeField]
	private bool isKey;

	[SerializeField]
	private Vector IsThisVector;
```